### PR TITLE
Move all config constants into single file.

### PR DESCRIPTION
**Pull request by robertnishihara**
**This PR does the following:**
- Creates a single file with all of the configuration parameters.
- Exposes all config parameters to Python (most of these are not used).

**Notes:**
- These parameters are NOT part of the API. They are strictly for debugging purposes (e.g., turning off heartbeat timeouts.
- We should do the same thing with other global constants like task scheduling states and NIL IDs.
- This does NOT pass any config parameters into the object store, even though some of those constants are defined here.
- This PR uses the same naming convention for the constants in C++ and Python... thoughts about that?

### CODE REVIEW

File: src/local_scheduler/local_scheduler_extension.cc [hunk 1]
```diff
@@ -287,6 +292,14 @@ MOD_INIT(liblocal_scheduler_library) {
   Py_INCREF(LocalSchedulerError);
   PyModule_AddObject(m, "local_scheduler_error", LocalSchedulerError);
 
+  Py_INCREF(&PyRayConfigType);
+  PyModule_AddObject(m, "RayConfig", (PyObject *) &PyRayConfigType);
+
+  /* Create the global config object. */
+  PyObject *config = PyRayConfig_make();
+  /* TODO(rkn): Do we need Py_INCREF(config)? */
```
Comment (robertnishihara) [inline src/local_scheduler/local_scheduler_extension.cc]: Might need to do this.

File: src/common/lib/python/common_extension.h [hunk 2]
```diff
@@ -49,8 +49,6 @@ PyObject *check_simple_value(PyObject *self, PyObject *args);
 PyObject *PyTask_to_string(PyObject *, PyObject *args);
 PyObject *PyTask_from_string(PyObject *, PyObject *args);
 
-PyObject *compute_put_id(PyObject *self, PyObject *args);
-
```
Comment (robertnishihara) [inline src/common/lib/python/common_extension.h]: Removed because this was unused.

**Review by pcmoritz — APPROVED** [review]

File: src/common/io.h [hunk 3]
```diff
@@ -6,16 +6,6 @@
 
 #include <vector>
 
-#define RAY_PROTOCOL_VERSION 0x0000000000000000
-
-/* Number of times we try binding to a socket. */
-#define NUM_BIND_ATTEMPTS 5
-#define BIND_TIMEOUT_MS 100
```
Comment (robertnishihara) [inline src/common/io.h]: `NUM_BIND_ATTEMPTS` and `BIND_TIMEOUT_MS` were not used anywhere.

**Comment by robertnishihara** [comment]
cc @imzhenyu

This moves all (unless I missed some), configuration constants into a single file `src/common/state/ray_config.h`. It still requires recompilation to change the constants. Later, we should make it possible to set these when Ray is started.

**Pull request merged**